test/Npgsql.Tests/Types/CompositeTests.cs

```diff
@@ -810,5 +810,71 @@ public void LocalMappingDontLeak()
                 }
             }
         }
+
+        [Test]
+        public void ThrowsOnWriteWithoutGetter()
+        {
+            var csb = new NpgsqlConnectionStringBuilder(ConnectionString)
+            {
+                Pooling = false,
+                ApplicationName = nameof(ThrowsOnWriteWithoutGetter)
+            };
+
+            try
+            {
+                using var connection = OpenConnection(csb);
+                connection.ExecuteNonQuery("CREATE TYPE composite_without_getter AS (value int)");
+                connection.ReloadTypes();
+                connection.TypeMapper.MapComposite<CompositeWithoutGetter>();
+
+                using var command = new NpgsqlCommand("SELECT @p", connection);
+                command.Parameters.AddWithValue("p", new CompositeWithoutGetter());
+
+                Assert.That(() => command.ExecuteScalar(), Throws.TypeOf<NotSupportedException>());
+            }
+            finally
+            {
+                using var conn = OpenConnection(csb);
+                conn.ExecuteNonQuery("DROP TYPE IF EXISTS composite_without_getter");
+            }
+        }
+
+        class CompositeWithoutGetter
+        {
+            public int Value { set { } }
```

Comment:
I don't think it affects anything you're testing, but did you mean to avoid the auto-setter?

Reply:
There is no write-only auto property (: You can write only two types of them: read-only and read-write.

Reply:
Oh, that makes sense! I've never tried to use one (clearly), but just assumed they were out there somewhere haha.